### PR TITLE
Better handling of struct pointers

### PR DESCRIPTION
**Pull request by epipho**
- `print structptr.member` now works
- attempting to print a pointer variable that is nil will no longer panic
- attempting to print a member variable of a nil struct will no longer panic
- added a bunch more tests for error conditions

### CODE REVIEW

File: proctl/variables_test.go [hunk 1]
```diff
@@ -35,26 +37,31 @@ func TestVariableEvaluation(t *testing.T) {
...
+		{"i8", "1", "int8", nil},
+		{"f32", "1.2", "float32", nil},
+		{"a6.Baz", "8", "int", nil},
+		{"a7.Baz", "5", "int", nil},
+		{"a8.Baz", "feh", "struct string", nil},
+		{"a9.Baz", "nil", "int", errors.New("a9 is nil")},
+		{"a9.NonExistent", "nil", "int", errors.New("a9 has no member NonExistent")},
+		{"a8", "main.FooBar2 {Bur: 10, Baz: feh}", "main.FooBar2", nil}, // reread variable after member
+		{"i32", "[2]int32 [1 2]", "[2]int32", nil},
+		{"NonExistent", "", "", errors.New{"could not find symbol value for NonExistent"}}
```
Comment (derekparker) [inline proctl/variables_test.go]: Missing ending comma here, won't compile.

File: proctl/variables.go [hunk 2]
```diff
@@ -490,27 +498,71 @@ func (thread *ThreadContext) extractVariableFromEntry(entry *dwarf.Entry) (*Vari
...
+}
 
-		fctx := fde.EstablishFrame(regs.PC())
-		cfaOffset := fctx.CFAOffset() + int64(regs.SP())
+// Extracts the value from the instructions given in the DW_AT_location entry.
+// We execute the stack program described in the DW_OP_* instruction stream, and
+// then grab the value from the other processes memory.
+func (thread *ThreadContext) extractValue(instructions []byte, addr int64, typ interface{}) (string, error) {
+	var err error
+	address := addr
```
Comment (derekparker) [inline proctl/variables.go]: Might as well just use `addr` here. It isn't used after being assigned to the local, so we shouldn't have to preserve it.

File: proctl/variables_test.go (hunk 1 above)
Comment (epipho) [inline proctl/variables_test.go]: Whoops, had that fixed locally and didn't commit. *hangs head in shame*

**Comment by derekparker** [comment]
Thanks, if you don't mind squashing this, I'll merge it in.

**Comment by epipho** [comment]
Squashed.

**Comment by derekparker** [comment]
Merged.

**Pull request closed without merging**